core/iwasm/common/wasm_runtime_common.c

```diff
@@ -2190,6 +2190,13 @@ wasm_set_exception(WASMModuleInstance *module_inst, const char *exception)
     }
 
 #if WASM_ENABLE_THREAD_MGR != 0
+
+#if WASM_ENABLE_SHARED_MEMORY
+    if (exception) {
+        notify_stale_threads_on_exception(
+            (WASMModuleInstanceCommon *)module_inst);
```

Comment:
Is this PR to terminating the thread which is atomic waiting, when there is other thread trapping of proc_exiting? Is it duplicated with https://github.com/bytecodealliance/wasm-micro-runtime/pull/1902? Seems #1902 can trigger the signal handler of the atomic waiting thread when os_pthread_kill to it, and in the signal handler, the atomic waiting thread can longjmp into the place of setjmp before, and exit normally?

Reply:
Suppose the patch can notify the current atomic wait operation, but if here is another atomic wait operation followed, will the thread be blocked again?

Reply:
@wenyongh did you mean #1903?

Reply:
Oh, sorry, I meant #1930, but it only works for posix platforms now, maybe this PR is also necessary.